### PR TITLE
[docs] Provide information about resource deadlocks, early stopping in Tune docs

### PR DESCRIPTION
**Pull request by Yard1**
## Why are these changes needed?

Provides a section in the FAQ regarding deadlocks caused by not enough resources being allocated to trials, and extends the "Which search algorithm/scheduler should I choose?" section

## Related issue number

Related to https://github.com/ray-project/ray/issues/18808, https://github.com/modin-project/modin/issues/3479#issuecomment-927864914

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by Yard1** [comment]
CC @devin-petersohn @jmakov

**Comment by jmakov** [comment]
Much clearer also for newcomers. Thanks!

**Review by devin-petersohn — COMMENTED** [review]
LGTM, thanks @Yard1 !

File: doc/source/tune/_tutorials/_faq.inc [hunk 1]
```diff
@@ -33,14 +41,19 @@ Alternatively, you can use :ref:`Population Based Training <tune-scheduler-pbt>`
 works well with few trials, e.g. 8 or even 4. However, this will output a hyperparameter *schedule* rather
 than one fixed set of hyperparameters.
 
-**If you have a small number of hyperparameters**, Bayesian Optimization-methods
-work well. Take a look at :ref:`BOHB <tune-scheduler-bohb>` to combine the
-benefits of bayesian optimization with early stopping.
+**If you have a small number of hyperparameters**, Bayesian Optimization methods
+work well. Take a look at :ref:`BOHB <tune-scheduler-bohb>` or :ref:`Optuna <tune-optuna>`
+with :ref:`ASHA <tune-scheduler-hyperband>` scheduler to combine the
```
Comment (krfricke) [inline doc/source/tune/_tutorials/_faq.inc:46]:
```suggestion
with the :ref:`ASHA <tune-scheduler-hyperband>` scheduler to combine the
```

File: doc/source/tune/_tutorials/_faq.inc [hunk 2]
```diff
@@ -33,14 +41,19 @@ Alternatively, you can use :ref:`Population Based Training <tune-scheduler-pbt>`
...
+benefits of Bayesian Optimization with early stopping.
 
 **If you only have continuous values for hyperparameters** this will work well
-with most Bayesian-Optimization methods. Discrete or categorical variables still
+with most Bayesian Optimization methods. Discrete or categorical variables still
 work, but less good with an increasing number of categories.
 
+**If you have many categorical values for hyperparemeters**, consider using random search,
```
Comment (krfricke) [inline doc/source/tune/_tutorials/_faq.inc:53]:
```suggestion
**If you have many categorical values for hyperparameters**, consider using random search,
```

**Review by krfricke — APPROVED** [review]

**Pull request merged**